### PR TITLE
[docs] Go documentation restructure

### PR DESCRIPTION
**Pull request by bmorelli25**
For https://github.com/elastic/apm-agent-go/issues/681. Demo available **[here](http://apm-agent-go_682.docs-preview.app.elstc.co/diff)**.

- [x] Top level heading and doc sync for log correlation: https://github.com/elastic/apm/issues/149
- [x] Use shared version attributes: https://github.com/elastic/apm/issues/150
- [x] Add a section on how the Agent does instrumentation: https://github.com/elastic/apm/issues/146
- [x] Rename the APM UI to "APM app": https://github.com/elastic/apm-dev/issues/567
- [x] Use `apm-app-ref`: https://github.com/elastic/docs/pull/1276
- [x] Updates to the new documentation structure outlined [here](https://docs.google.com/spreadsheets/d/1P53LbNY7jFSLLJhOKw-nmDenQbrIFlY4EEFJGqT1kmY/edit#gid=0)
- [x] Sync Central config docs: https://github.com/elastic/apm-agent-dotnet/pull/497#discussion_r326303325
- [x] Add placeholder .asciidoc file for release notes - https://github.com/elastic/docs/pull/1502

### CODE REVIEW

File: docs/introduction.asciidoc [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
+https://golang.org/pkg/database/sql/[database/sql] drivers.
+The Agent also offers an <<api>> for custom instrumentation.
+
+[float]
+[[how-it-works]]
+=== How does the Agent work?
+
+The Agent auto-instruments <<supported-tech>> and records interesting events,
+like spans for database queries and transactions for incoming HTTP requests.
+To do this, it [[NEED_CONTENT_HERE]]
```
Comment (bmorelli25) [inline docs/introduction.asciidoc]: @axw, I'll need your help with this one. Can you jot down some words about how the Go Agent works? Nothing too deep, just a high level overview. Inspiration may be found [here](https://www.elastic.co/guide/en/apm/agent/python/current/getting-started.html#how-it-works) or [here](https://www.elastic.co/guide/en/apm/agent/ruby/current/introduction.html#how-it-works).

File: docs/instrumenting.asciidoc [hunk 2]
```diff
@@ -1,17 +1,15 @@
 [[getting-started]]
```
Comment (bmorelli25) [inline docs/instrumenting.asciidoc]: How do you feel about the changes here? There is a lot going on [here](https://www.elastic.co/guide/en/apm/agent/go/current/instrumenting-source.html), and I think it'd be useful to break things up a little bit. These changes create a new page that shows, briefly, the three steps required to get started. Pages for modules and custom instrumentation are nested underneath.

Screen Shot 2019-11-27 at 1 42 13 PM

File: docs/introduction.asciidoc (hunk 1 above)
Comment (axw) [reply docs/introduction.asciidoc]: How's this?

> The Agent includes instrumentation modules for `<<supported-tech>>`, each providing middleware or wrappers for recording interesting events, such as incoming HTTP requests, outgoing HTTP requests, and database queries.
>
> To collect data about incoming HTTP requests, you can install router middleware for one of the supported `<<supported-tech-web-frameworks>>`. Incoming requests will be recorded as transactions, along with any related panics or errors.
>
> To collect data for outgoing HTTP requests, you can instrument an `http.Client` or `http.Transport` using `<<builtin-modules-apmhttp>>`. To collect data about database queries, you can use `<<builtin-modules-apmsql>>`, which provides instrumentation for well known database drivers.
>
> In order to connect transactions with related spans and errors, and propagate traces between services (distributed tracing), the agent relies on Go's built-in [context](https://golang.org/pkg/context/) package: transactions and spans are stored in context objects. For example, for incoming HTTP requests in-flight trace data will be recorded in the `context` object accessible through [net/http.Context](https://golang.org/pkg/net/http/#Request.Context). You can read more about this in `<<custom-instrumentation-propagation>>`.
>
> In addition to capturing events like those mentioned above, the agent also also collects system and application metrics at regular intervals. This collection happens in a background goroutine that is automatically started when the agent is initialized.

File: docs/instrumenting.asciidoc (hunk 2 above)
Comment (axw) [reply docs/instrumenting.asciidoc]: This is much nicer, thanks.

I think we should probably pull Context Propagation out of Custom Instrumentation, and have it at the same level. There's two aspects to instrumentation: instrumenting individual components (e.g. HTTP server, database driver), and then "context propagation" - connecting up the dots.

File: docs/introduction.asciidoc (hunk 1 above)
Comment (bmorelli25) [reply docs/introduction.asciidoc]: Awesome! Thank you

File: docs/instrumenting.asciidoc (hunk 2 above)
Comment (bmorelli25) [reply docs/instrumenting.asciidoc]: Thank you. I pulled context propagation out of instrumentation and added a note about it in the setup section.

**Comment by bmorelli25** [comment]
Rerequested a review from you. Not sure if you want to take another look at this before I merge.

**Pull request merged**